### ISSUE TITLE
Error result when loop with Iterators.partition

### ISSUE THREAD

**Issue by bearxu83** [JuliaData/CSV.jl#387]
```
test_buffer = IOBuffer("a,b\n1,1\n2,2\n3,3\n")
seek(test_buffer, 0)
csv_test = CSV.File(test_buffer)
for rows in Iterators.partition(csv_test, 2)
    for row in rows
        # get last property
        # getproperty(row, :b)
        println(row.a)
    end
end
```
result is `1 2 3`

after uncomment the `get_property`
result becomes
`
missing
2
3
`

> In Windows 10, Julia 1.0.3

**Comment by bearxu83** [comment]
It seems it is the behavior of partition.

**Comment by quinnj** [comment]
This will be fixed in #395

**Comment by quinnj** [comment]
Fixed in #395
